protocol/v2/qbft/controller/controller.go

```diff
@@ -55,26 +54,26 @@ func NewController(
 }
 
 // StartNewInstance will start a new QBFT instance, if can't will return error
-func (c *Controller) StartNewInstance(logger *zap.Logger, height specqbft.Height, value []byte) error {
-
+func (c *Controller) StartNewInstance(logger *zap.Logger, height specqbft.Height, value []byte) (*instance.Instance, error) {
 	if err := c.GetConfig().GetValueCheckF()(value); err != nil {
-		return errors.Wrap(err, "value invalid")
+		return nil, errors.Wrap(err, "value invalid")
 	}
 
 	if height < c.Height {
-		return errors.New("attempting to start an instance with a past height")
+		return nil, errors.New("attempting to start an instance with a past height")
 	}
 
 	if c.StoredInstances.FindInstance(height) != nil {
-		return errors.New("instance already running")
+		return nil, errors.New("instance already running")
 	}
 
 	c.Height = height
 
 	newInstance := c.addAndStoreNewInstance()
 	newInstance.Start(logger, value, height)
 	c.forceStopAllInstanceExceptCurrent()
-	return nil
+
+	return newInstance, nil
 }
```

Comment:
i agree this is an improvement to avoid querying after creating, but since this code is basically a duplication from the spec (for the time being), i think we should stick to being closely aligned to make it easier to diff and bring over future changes they make

https://github.com/ssvlabs/ssv-spec/blob/5b5e3dc5e572d1e50bc5503f2cd68b401ed925a9/ssv/runner.go#L244-L253

this is true only for the `qbft` and `runner` packages, in the rest of the packages we're free to make any improvements we want